src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -135,6 +136,15 @@ void SyncInferRequest::infer() {
 
     push_input_data();
 
+    MemoryControl* network_memory_control = m_graph->getGraphContext()->getMemoryControl();
+    if (!network_memory_control) {
+        OPENVINO_THROW("Memory control unit is not initilized for graph: ", m_graph->GetName());
+    }
+
+    if (!network_memory_control->allocated()) {
+        network_memory_control->allocateMemory();
+    }
```

Comment:
I don't really remember why I moved it out the Graph, but I think it can be reverted.
From the other perspective we kind of know that we don't have to perform this check for the SubGraphs, so it kind of make sense to move the check out of the Graph logic.